src/cryptography/fernet.py

```diff
@@ -30,7 +30,13 @@ def __init__(
         key: typing.Union[bytes, str],
         backend: typing.Any = None,
     ):
-        key = base64.urlsafe_b64decode(key)
+        try:
+            key = base64.urlsafe_b64decode(key)
+        except binascii.Error as exc:
+            raise ValueError(
+                f"Fernet key must be 32 url-safe base64-encoded bytes, "
+                f"not {key!r}."
```

Comment:
Putting the key value in the exception is a dangerous thing to do since there it dramatically increases the odds sensitive material will get accidentally logged at some point. Otherwise I'm okay with this, but you'll need to add a test that exercises this path or else coverage will fail.

Reply:
I like your thinking. Fixing as we speak.